src/pgwire/src/protocol.rs

```diff
@@ -243,6 +254,30 @@ where
         .vars_mut()
         .end_transaction(EndTransactionAction::Commit);
 
+    if !session.user().is_internal() && !session.user().is_external_admin() {
```

Comment:
This logic also needs to live somewhere in http, so should probably be a method on ConnectionCounter that takes a Session.

Reply:
Chatted offline. I think the plan is to leave this in. Happy to revisit when adding connection caps to http